### DISCUSSION TITLE
Why is BASE_URL in datasource config deprecated

### DISCUSSION THREAD

**Discussion by magaton** [grafana/grafana-infinity-datasource#212] [General]
It is very convenient to specify base url that one can override in a query.
This way if the absolute url is buried in queries, it is quite difficult to switch to an alternative graphql endpoint.

**Comment by yesoreyeram** [comment]
It was actually causing confusions when a user specify two different url in config and query . It is deprecated but no plans to remove.so it should work the way it was used to.

**Reply by canob** [reply]
Hi @yesoreyeram,
I think the Base URL is not working as expected if you try to use it, by not completing the URL in the Explore Panel:
If you not complete the URL in the Explore Panel, Infinity is going to use your Base URL + the example URL, https://raw.githubusercontent.com/grafana/grafana-infinity-datasource/main/testdata/users.json, + the configured URL Params:
![imagen](https://github.com/grafana/grafana-infinity-datasource/assets/30271927/2298b621-82bf-4351-835b-035d08aef1fe)
As a workaround, you can add a SPACE in the URL, and is going to work:
![imagen](https://github.com/grafana/grafana-infinity-datasource/assets/30271927/4d4f871c-dfa7-441f-a85a-de95335b747c)

**Reply by yesoreyeram** [reply]
captured in https://github.com/grafana/grafana-infinity-datasource/issues/908

**Comment by canob** [comment]
Another big problem of deprecating "Base URL": if you want to use a Grafana Datasource Variable with Infinity, for example, to switch between different SaaS products API sources in the same table, with this deprecation, you cannot, because you don´t have the URL configured previously in the datasource.
This is really annoying, and, at the end, it is opposed to the "natural" idea of a Data Source in Grafana, which I think it is that when you "reach" Explore in Grafana, you already have the authentication access to the Data Source configured (that includes headers, query params, and URL in this case), and you can start accessing the data immediately.

**Reply by canob** [reply]
And another problem, related to the one commented before, but in this case with the deprecation of Global Queries (is it really deprecated, as said in https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/references/global-queries/? Because there is not message about that in the Grafana UI, like the one that you can see in Misc/Base URL, but it is not working right now in 2.8.0 version): for the same scenario mentioned in the previous comment, If you want to switch between different SaaS products API sources in the same table/panel, that different products are going to have different outputs/fields/rows, so you need to "play" with them on "Parsing options & Result fields" of the Global Query to have a different parsing option for each different API product, and in this moment, you cannot, you have the same "Parsing options & Result fields" for every Infinity Datasource that you select using a DataSource Variable for the same panel.
